compiler/rustc_serialize/src/serialize.rs

```diff
@@ -50,6 +50,19 @@ pub trait Encoder {
         self.emit_usize(v_id);
         f(self);
     }
+
+    #[inline]
+    fn emit_big_u32(&mut self, v: u32) {
+        // Shift by 2 to give us space to encode how many bytes we need
+        let mut v = v << 2;
```

Comment:
I assume there's a todo here for handling `u32`s that are more than 30 bits?

Reply:
Yeah. Those would spill into a 5th byte which makes me grumpy and also question using this encoding scheme for them.

But also the niche for this type is odd (the high 2 bits, ye gods that's a big niche). `rustc_index` types by default have a 256-bit niche at the top of the range which sounds kind of useful but maybe not.